### PR TITLE
Fix owner not being checked on write to book

### PR DESCRIPTION
**Pull request by rubenwardy**
This is a security vulnerability, as the client can submit fake data.

### CODE REVIEW

File: mods/default/craftitems.lua [hunk 1]
```diff
@@ -84,6 +85,11 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 			data = minetest.deserialize(stack:get_metadata())
 		end
 
+		if data and data.owner and data.owner ~= player:get_player_name()  then
+			minetest.chat_send_player(player:get_player_name(), "You don't own this book!")
```
Comment (sfan5) [inline mods/default/craftitems.lua]: no need to chat msg if this can't happen during normal usage

File: mods/default/craftitems.lua [hunk 2]
```diff
@@ -84,6 +85,11 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 			data = minetest.deserialize(stack:get_metadata())
 		end
 
+		if data and data.owner and data.owner ~= player:get_player_name()  then
```
Comment (Thomas--S) [inline mods/default/craftitems.lua]: How about using `default.can_interact_with_node()` here? I don't know, it's just a suggestion.

Comment (rubenwardy) [inline mods/default/craftitems.lua]: it's a craft item, not a node. And that's out of scope

**Comment by rubenwardy** [comment]
https://github.com/minetest/minetest_game/commit/2917562728927eda925d3e9ba17bd3cf945dae2a

**Pull request closed without merging**